pubtools/_pulp/tasks/fix_cves.py

```diff
@@ -0,0 +1,167 @@
+import sys
+import logging
+import random
+import attr
+
+from pubtools.pulplib import Criteria, ErratumUnit, ErratumReference
+
+from pubtools._pulp.task import PulpTask
+from pubtools._pulp.services import PulpClientService
+from pubtools._pulp.tasks.common import Publisher
+from pubtools._pulp.arguments import SplitAndExtend
+from pubtools._pulp.tasks.push.items.erratum_conv import bump_erratum_version
+
+step = PulpTask.step
+
+LOG = logging.getLogger("pubtools.pulp")
+
+
+class FixCves(PulpTask, PulpClientService, Publisher):
+    """Command to fix cves"""
+
+    def fail(self, *args, **kwargs):
+        LOG.error(*args, **kwargs)
+        sys.exit(30)
+
+    @step("Get erratum")
+    def get_erratum(self, advisory_id):
+        erratum_page = self.get_erratum_from_pulp(advisory_id)
+        erratum = [ert for ert in erratum_page]
+        if not erratum:
+            self.fail("No erratum found for %s", advisory_id)
+        # multiple erratums should not exist
+        assert len(erratum) == 1
+
+        return erratum[0]
+
+    def get_erratum_from_pulp(self, advisory_id):
+        crit = Criteria.and_(
+            Criteria.with_unit_type(ErratumUnit), Criteria.with_id(advisory_id)
+        )
+        return self.pulp_client.search_content(criteria=crit).result()
+
+    @step("Process erratum")
+    def process_erratum_for_upload(self, erratum, cves):
+        # get non-cve erratum references
+        refs = [ref for ref in erratum.references if ref.type != "cve"]
+
+        # add new cves to erratum
+        for cve in cves:
+            refs.append(
+                ErratumReference(
+                    type="cve",
+                    title=cve,
+                    id=cve,
+                    href="https://www.redhat.com/security/data/cve/%s.html" % cve,
+                )
+            )
+
+        # updated version
+        version = bump_erratum_version(erratum)
+
+        # update erratum
+        # clear repository memberships. It will be evaluated by Pulp.
+        erratum = attr.evolve(
+            erratum, references=refs, repository_memberships=None, version=str(version)
+        )
+
+        LOG.debug("Processed erratum: %s", attr.asdict(erratum))
+
+        return erratum
+
+    @step("Upload erratum")
+    def upload_erratum(self, erratum, repos):
+        # find repo to upload
+        repo = self.get_upload_repo(repos)
+
+        # upload erratum to the repo
+        repo.upload_erratum(erratum).result()
+        LOG.info("Uploaded to %s", repo.id)
+
+    def get_repo_from_pulp(self, repo_id):
+        crit = Criteria.with_id(repo_id)
+        resp = self.pulp_client.search_repository(criteria=crit)
+
+        return [r for r in resp.result()]
+
+    def get_upload_repo(self, repos):
+        return random.choice(repos)
```

Comment:
I see that using random might pose a stability issue for tests and maybe this is why you mocked it.

May I suggest the following instead: rather than random.choice(), create a random.Random object with a controllable seed, e.g. like this:

```python
self.random = random.Random(os.getenv('PUBTOOLS_SEED') or random.random())
```

Then call self.random.choice.

Tests can then obtain results which are stable but still random by setting PUBTOOLS_SEED to some specific value before constructing an instance of the task, e.g. just '1'.  This also works fine from higher level tests (e.g. if there are some Pub tests which run these tasks).